vevestaX/vevesta.py

```diff
@@ -106,6 +106,28 @@ def endModelling(self):
     # create alias of method modellingStart and modellingEnd
     start = startModelling
     end = endModelling
+    
+    
+    # function to get arguments of a function
+    def parameters(self, func):
+        def wrapper(*args, **kwargs):
+            # to get parameters of function that are passed
+            functionParameters = inspect.signature(func).bind(*args, **kwargs).arguments
+            functionParameters = dict(functionParameters)
+            # to get values that are not passed and defautlt values
+            defaultParameters = inspect.signature(func)
+            for param in defaultParameters.parameters.values():
+                if ( param.default is not param.empty):
+                    # checks in key exist in abouve dictionary then doesn't update it will default value, otherwise append into dictionary
+                    if param.name not in functionParameters:
+                        functionParameters[param.name] = param.default
+            
+            for key, value in functionParameters.items():
+                if type(value) in [int, float, bool, str]:
+                    if key not in self.__variables:
```

Comment:
see if you can use some python construct to optimize this code...like list comprehension